### PR TITLE
udpate dependencies, code style and ci node versions

### PR DESCRIPTION
**Pull request by vdeturckheim**
This PR aims at closing :
* #29 with ES2015 updates according to hapi style rules
* #28 dependencies updated to latest versions

Also :
* update of travis target node versions

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -10,17 +10,17 @@
...
     "good": ">= 6.x.x"
   },
   "devDependencies": {
-    "code": "1.x.x",
-    "lab": "6.x.x"
+    "code": "2.x.x",
+    "lab": "10.x.x"
   },
   "engines": {
     "node": ">=0.10.x"
```
Comment (gergoerdosi) [inline package.json]: This needs to be changed to `">=4.0.0"` too.

File: .travis.yml [hunk 2]
```diff
@@ -2,5 +2,7 @@ language: node_js
 sudo: false
 
 node_js:
-  - 0.10
```
Comment (arb) [inline .travis.yml]: Match what hapi core uses here - https://github.com/hapijs/hapi/blob/master/.travis.yml

File: lib/index.js [hunk 3]
```diff
@@ -101,7 +104,7 @@ internals.GoodHttp.prototype._sendMessages = function () {
     envelope.events = this._settings.mapEvents ?
             internals.createEventMap(this._eventQueue) : this._eventQueue;
 
-    var wreckOptions = {
+    const wreckOptions = {
```
Comment (arb) [inline lib/index.js]: On line 111, you might be able to get away from `Hoek.merge` and instead use `Object.assign`

File: lib/index.js [hunk 4]
```diff
@@ -101,7 +104,7 @@ internals.GoodHttp.prototype._sendMessages = function () {
...
         payload: Stringify(envelope)
     };
```
Comment (arb) [inline lib/index.js]: On line 111, you might be able to get away from `Hoek.merge` and instead use `Object.assign`

**Comment by arb** [comment]
Can you please reword you commit so it says "Closes #29. Closes #28" That way the associated issues will automatically close when this gets merged in.

File: package.json [hunk 5]
```diff
@@ -10,17 +10,17 @@
   },
   "dependencies": {
     "good-squeeze": "2.x.x",
-    "hoek": "2.x.x",
+    "hoek": "3.x.x",
     "json-stringify-safe": "5.0.x",
-    "lodash.groupby": "3.0.x",
-    "wreck": "5.x.x"
+    "lodash.groupby": "4.2.x",
```
Comment (arb) [inline package.json]: Can you pin this to `4.2.0`

**Comment by vdeturckheim** [comment]
I probably lost the track of my branching. Sorry, closing this to do it properly.

**Pull request closed without merging**